packages/@sanity/core/src/util/checkRequiredDependencies.ts

```diff
@@ -0,0 +1,238 @@
+import path from 'path'
+import execa from 'execa'
+import semver from 'semver'
+import resolveFrom from 'resolve-from'
+import {readJSON, access} from 'fs-extra'
+import oneline from 'oneline'
+
+import {CliCommandContext} from '../types'
+
+/**
+ * Checks that the studio has declared and installed the required dependencies
+ * needed by the Sanity modules. While we generally use regular, explicit
+ * dependencies in modules, there are certain dependencies that are better
+ * served being peer dependencies, such as react and styled-components.
+ *
+ * If these dependencies are not installed/declared, we want to prompt the user
+ * whether or not to add them to `package.json` and install them using yarn/npm
+ */
+export async function checkRequiredDependencies(context: CliCommandContext): Promise<void> {
+  const {workDir: studioPath, output} = context
+  const [
+    basePeerDependencies,
+    studioPackageManifest,
+    installedStyledComponentsVersion,
+  ] = await Promise.all([
+    await readBasePeerDependencies(studioPath),
+    await readPackageManifest(path.join(studioPath, 'package.json')),
+    await readModuleVersion(studioPath, 'styled-components'),
+  ])
+
+  // If the `@sanity/base` package.json does not have a `styled-components` peer dependency
+  // declared, the user is probably running an old version of `@sanity/base`. This is a bit
+  // of an indeterminate state, so we'll just have to accept it and assume things will work.
+  const wantedStyledComponentsVersionRange = basePeerDependencies['styled-components']
+  if (!wantedStyledComponentsVersionRange) {
+    return
+  }
+
+  // The studio _must_ now declare `styled-components` as a dependency. If it's not there,
+  // we'll want to automatically _add it_ to the manifest and tell the user to reinstall
+  // dependencies before running whatever command was being run
+  const declaredStyledComponentsVersion = studioPackageManifest.dependencies['styled-components']
+  if (!declaredStyledComponentsVersion) {
+    const deps = {'styled-components': wantedStyledComponentsVersionRange}
+    await installDependenciesWithPrompt(deps, context)
+    return
+  }
+
+  // The declared version should be semver-compatible with the version specified as a
+  // peer dependency in `@sanity/base`. If not, we should tell the user to change it.
+  //
+  // Exception: Ranges are hard to compare. `>=5.0.0 && <=5.3.2 || ^6`... Comparing this
+  // to anything is going to be challenging, so only compare "simple" ranges/versions
+  // (^x.x.x / ~x.x.x / x.x.x)
+  if (
+    isComparableRange(declaredStyledComponentsVersion) &&
+    !semver.satisfies(
+      normalizeVersion(declaredStyledComponentsVersion),
```

Comment:
i think `normalizeVersion` could be replaced with `semver.minVersion`

Reply:
Ah, beautiful. Thanks.